docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -245,3 +245,92 @@ If you [self-host a Prefect server instance](/3.0rc/manage/self-host), run the f
 ```bash
 prefect server database upgrade -y
 ```
+
+## Flow run final state determination
+
+In Prefect 2, the final state of a flow run was influenced by the states of its task runs, i.e. if any task run failed, the flow run was marked as failed.
+
+In Prefect 3, the final state of a flow run is entirely determined by:
+
+1. The `return` value of the flow function (same as in Prefect 2):
+   - Literal values are considered successful.
+   - Any explicit `State` that is returned will be considered the final state of the flow run.
```

Comment:
It might also be worthwhile to include that if multiple states are returned (via a list or tuple) they all must be completed to be considered completed. If any are failed, then the final flow state is failed.